upgrade: upgrade utdanning-model to 4.0.0

daniejoh opened the pull request:
Can be merged to main instead, if 3.20 branch is merged.

Commit:
test: make tests compile and run

Commit:
upgrade: upgrade resource model to 3.20

Commit:
upgrade: Upgrade gradle and other necessary packages

Commit:
ci: updates CI to manual deployment

Commit:
upgrade: upgrade utdanning-model to 4.0.0

Upgrades utdanning-resource-model to 4.0.0
Some breaking changes in the upgrade, so had to change the check for
valid periode.
Also rewrote the service and test to test at a fixed date.

nozoz commented:
Ser bra ut! Eneste nit-picken jeg har er ved bruk av kommentarer. Jeg føler bruken av klasse kommentarer og noen metoder er veldig bra - men kanskje litt for mange i metodene? Jeg kan kommentere på noen eksempler, også vil jeg gjerne høre hva du mener.

nozoz commented on src/main/java/no/fint/kulturtanken/model/Trinn.java:
```diff
@@ -7,5 +7,5 @@
 public class Trinn {
 	private String niva;
-	private List<Basisgruppe> basisgrupper;
+	private List<Klasse> basisgrupper;
```
basisgrupper -> klasser?

daniejoh replied:
Hadde glemt dette flere steder. Har renamet alle nå :)

nozoz commented on src/main/java/no/fint/kulturtanken/service/KulturtankenService.java:
```diff
@@ -30,22 +34,37 @@ public class KulturtankenService {
+     */
     @Cacheable(value = "schoolOwner", unless = "#result == null")
     public Skoleeier getSchoolOwner(String orgId) {
+        // Fetch school owner from NSR
```
Mener dette er beskrivene i metode-kallet under.

nozoz commented on src/main/java/no/fint/kulturtanken/service/KulturtankenService.java:
```diff
@@ -30,22 +34,37 @@ public class KulturtankenService {
         if (schoolOwner == null) return null;
 
+
+        // If School already exist in kulturtankenProperties, fetch schools from NSR
```
Mener vi bør refaktorere koden til å fortelle oss hva som skjer istdenfor å bruke kommentarer. For eksempel kunne vi flyttet denne logikken:
```java
if (kulturtankenProperties.getOrganisations().get(orgId).getSource().equals("nsr")) {
}
```

til en egen metode slik:
```java
private boolean schoolAlreadyExistsInKulturtankenProperties(args...) {
    return kulturtankenProperties.getOrganisations()
        .get(orgId)
        .getSource()
        .equals("nsr");
}
```

nozoz commented on src/main/java/no/fint/kulturtanken/service/KulturtankenService.java:
```diff
@@ -30,22 +34,37 @@ public class KulturtankenService {
             return schoolOwner;
         }
 
+        // Fetch schools from FINT and populate with levels and subjects
```
Igjen føler jeg dette bør bo i en egen metode som beskriver det kommentaren sier. Men det er bedre enn det som var der før, så fortsatt veldig bra at du la det til!

nozoz reviewed (approved):
Veldig bra! 🫶

Commit:
chore: rename all basis groups to klasse

There was missing some renaming to klasse in variablenames and similar.
This commit also removes some unnecessary comments.

daniejoh committed:
Merge pull request #38 from FINTLabs/upgrade/CT-2215-upgrade-model-to-3.20

Upgrade/ct 2215 upgrade model to 3.20

Commit:
ci: upgrade gradle version(s)

Upgrade gradle versions in docker and actions

daniejoh committed:
Merge pull request #41 from FINTLabs/ci/upgrade-gradle-build-versions

ci: upgrade gradle version(s)

Commit:
ci: update azure login version

daniejoh committed:
Merge pull request #42 from FINTLabs/ci/upgrade-azure-login-version

ci: update azure login version

Commit:
ci: change matrix to inputs for getting environment

Commit:
ci: add upper-casing of environment

daniejoh committed:
ci: change method of uppercasing environmentname and retrieving it

daniejoh committed:
ci: add debug print for secret_name

daniejoh committed:
ci: debug hardcode secret name

daniejoh committed:
ci: add debug workflow

daniejoh committed:
ci: adds inherit secrets

daniejoh committed:
ci: add missing secret lookup

daniejoh committed:
ci: remove debug print

daniejoh committed:
fix: revert new authentication method

daniejoh committed:
Merge pull request #43 from FINTLabs/fix/revert-new-authentication-method

fix: revert new authentication method

jflatby committed:
Update imports to use `no.novari.fint` package and upgrade Fint resource model dependency to version 4.0.20.

jflatby committed:
Merge remote-tracking branch 'origin/main' into upgrade/CT-2215-upgrade-model-to-4.x

# Conflicts:
#	.github/workflows/CI.yaml
#	.github/workflows/manual-deployment-all.yaml
#	.gitignore
#	build.gradle
#	src/main/java/no/fint/kulturtanken/configuration/SecurityConfiguration.java

jflatby committed:
Fix typo in resource mapping key (KlasseResource -> KlasseResources).

jflatby committed:
Re-add deprecated password grant type in OAuth2AuthorizedClientProvider configuration.

jflatby committed:
Undo fix resource mapping key (KlasseResources -> KlasseResource).

jflatby committed:
Undo incorrect resource mapping key change (KlasseResource -> KlasseResources).

jflatby committed:
Handle errors gracefully in `getContent` by skipping failing registrations and logging warnings.

jflatby committed:
temp: test using basisgruppe instead of klasse

jflatby committed:
Undo resource mapping key for `KlasseResources` and error handling in `getContent`.

jflatby committed:
Add default header for model version override in WebClient configuration

jflatby committed:
Fix header casing for model version override in WebClient configuration
